backend/src/process.py

```diff
@@ -118,20 +119,29 @@ async def run_iteration(self, index: int, total: int):
                 "event": "iterator-progress-update",
                 "data": {
                     "percent": index / total,
+                    "index": index,
+                    "total": total,
+                    "eta": self.avg_time * (total - index),
                     "iteratorId": self.iterator_id,
                     "running": list(self.chain.nodes.keys()),
                 },
             }
         )
 
         try:
+            start = time.time()
             await executor.run_iteration(self.chain)
+            end = time.time()
+            self.avg_time = (self.avg_time * index + (end - start)) / (index + 1)
```

Comment:
Since indexes may be repeated, so this is incorrect. I think it would be better to use a list of durations instead. This would also make it possible to use something like exponential smoothing to give more weight to recent durations.